server/todo-api/controllers/controller.ts

```diff
@@ -0,0 +1,72 @@
+import { Response, NextFunction, Request } from 'express';
+import { ToDoDBModel } from '../schemas/to-do-schema';
+import { ToDoModel } from '../model/todo-model';
+import { Types } from 'mongoose';
+import { HTTPError } from '../types';
+import validator from 'validator';
+
+export class Controller {
+  async create(req: Request, res: Response, next: NextFunction): Promise<void> {
+    const errors: string[] = [];
+
+    const { createdBy, isCompleted, description, priority, title, dueDate } =
+      req.body as ToDoModel;
```

Comment:
No es correcto hacer un cast directamente sin haber verificado antes que los atributos sean correctos. Debemos crear un [typeGuard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) y verificar que los atributos esten dentro del objeto.
```suggestion
      req.body;
```